test/internal/ban-words.test.ts

```diff
@@ -1,91 +1,108 @@
+banWord("std.mem.indexOfAny(u8", "Use bun.strings.indexOfAny", 2);
+banWord("std.StringArrayHashMapUnmanaged(", "bun.StringArrayHashMapUnmanaged has a faster `eql`", 12);
+banWord("std.StringArrayHashMap(", "bun.StringArrayHashMap has a faster `eql`", 1);
+banWord("std.StringHashMapUnmanaged(", "bun.StringHashMapUnmanaged has a faster `eql`");
+banWord("std.StringHashMap(", "bun.StringHashMap has a faster `eql`");
+banWord("std.enums.tagName(", "Use bun.tagName instead", 2);
+banWord("std.unicode", "Use bun.strings instead", 33);
 
-  [String.raw`: [a-zA-Z0-9_\.\*\?\[\]\(\)]+ = undefined,`]: { reason: "Do not default a struct field to undefined", limit: 241, regex: true },
-  "usingnamespace": { reason: "Zig 0.15 will remove `usingnamespace`" },
+const allocator_ptr_ban_msg =
+  "The std.mem.Allocator context pointer can be undefined, which makes this comparison undefined behavior";
+banWord("allocator.ptr ==", allocator_ptr_ban_msg);
+banWord("allocator.ptr !=", allocator_ptr_ban_msg, 1);
+banWord("== allocator.ptr", allocator_ptr_ban_msg);
+banWord("!= allocator.ptr", allocator_ptr_ban_msg);
+banWord("alloc.ptr ==", allocator_ptr_ban_msg);
+banWord("alloc.ptr !=", allocator_ptr_ban_msg);
+banWord("== alloc.ptr", allocator_ptr_ban_msg);
+banWord("!= alloc.ptr", allocator_ptr_ban_msg);
 
-  "std.fs.Dir": { reason: "Prefer bun.sys + bun.FD instead of std.fs", limit: 180 },
-  "std.fs.cwd": { reason: "Prefer bun.FD.cwd()", limit: 103 },
-  "std.fs.File": { reason: "Prefer bun.sys + bun.FD instead of std.fs", limit: 64 },
-  ".stdFile()": { reason: "Prefer bun.sys + bun.FD instead of std.fs.File. Zig hides 'errno' when Bun wants to match libuv", limit: 18 },
-  ".stdDir()": { reason: "Prefer bun.sys + bun.FD instead of std.fs.File. Zig hides 'errno' when Bun wants to match libuv", limit: 48 },
+banWord(/: [a-zA-Z0-9_.*?[\]()]+ = undefined,/g, "Do not default a struct field to undefined", 241);
+banWord("usingnamespace", "Zig 0.15 will remove `usingnamespace`");
 
-  ".arguments_old(": { reason: "Please migrate to .argumentsAsArray() or another argument API", limit: 289 },
+const prefer_bun_reason =
+  "Prefer bun.sys + bun.FD instead of std.fs.File. Zig hides 'errno' when Bun wants to match libuv";
+banWord("std.fs.Dir", prefer_bun_reason, 180);
+banWord("std.fs.cwd", "Prefer bun.FD.cwd()", 103);
+banWord("std.fs.File", prefer_bun_reason, 64);
+banWord(".stdFile()", prefer_bun_reason, 18);
+banWord(".stdDir()", prefer_bun_reason, 48);
 
-  "// autofix": { reason: "Evaluate if this variable should be deleted entirely or explicitly discarded.", limit: 176 },
-};
-const words_keys = [...Object.keys(words)];
+banWord(".arguments_old(", "Please migrate to .argumentsAsArray() or another argument API", 289);
+banWord("// autofix", "Evaluate if this variable should be deleted entirely or explicitly discarded.", 176);
+banWord("catch unreachable", "Prefer handling error, or using catch bun.outOfMemory() for OutOfMemory errors.", 1857);
```

Comment:
does not handle a common pattern where the unreachable is on its own line.

![CleanShot 2025-04-25 at 17 20 22@2x](https://github.com/user-attachments/assets/24a4d2c4-427b-446c-a3c0-72c1cb1766f6)

however, some of these uses are correct, such as the ones in DevServer. the ban word might be able to be done in a way where it omits the commented cases.

Reply:
ban-words doesn't support multiline regex right now so it can't really catch that one.

Of all the inline 'catch unreachable's, there's three with a comment justifying their reason to exist and three with an unrelated comment. I could switch it to `/catch unreachable(?!;\s*\/\/)/g`

Two of the unreachables in DevServer.zig are actually unreachable, but `ev.timer = std.time.Timer.start()` could throw if clock_gettime fails and it would probably be better to crash in that case than be UB.